gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineImmutables.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import java.util.Objects;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+public final class BaselineImmutables implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().withPlugin("java", unused -> {
+            project.afterEvaluate(proj -> {
```

Comment:
Thanks for tracking down this immutables behaviour and building this plugin! I'm excited for the speedup it should give repos.

Just made a follow up PR to remove this use of afterEvaluate - it's technically deprecated (although I don't think Gradle will ever be able to remove it) and the proliferation of afterEvaluates has caused us issues in the past to do with the order they run in. Luckily, we can use some of the newer lazy APIs to avoid this: https://github.com/palantir/gradle-baseline/pull/1752

Reply:
Ah sorry, thanks for fixing this! Didn't know that `afterEvaluate` was deprecated. Wasn't sure when this should be applied so I just copied code from an existing plugin.